docs/contributing/mentorship.md

```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
+* Try to have some prerequisites in mind while shortlisting an application of a potential mentee (CV/questions/proposal/interview),
+  * The mentee should be able to allocate enough time to the mentorship.
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and generally not being able to utilize the time mentors can spend with them.
+    * So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution.
+  * The mentee should have at least one prior contribution to Thanos or its dependencies that affect it indirectly. The contribution should demonstrate effort & interest on part of the potential mentee.
+    * We want to understand that the potential mentee is interested in the project and has invested some time and effort in trying to understand Thanos.
+    * However, the contribution should not be a minimal effort, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort, and should instead be some code/doc change. Note that this doesn't need to be in a merged/finalized state and can be open.
+  * The mentee should have proper means of communication (can be ensured during the 20-minute interview).
+    * This constitutes things like good enough internet to be able to video calls without dropping, as that is the primary means of communication between mentors and mentees.
+    * As most of the discussions (meetings/issues/reviews/code comments) are conducted in English, the mentee should be able to communicate well in English (first/second language).
+  * The mentee should have some form of local development environment setup, that would not require the mentor to constantly debug the mentee's environment, while working on the project.
+    * This is hard to ensure fully, but generally the mentee should be working on linux/linux VM/macOS, so most generic GNU/BSD commands are compatible.
+    * The mentee should have basic git, bash, programming, and Go knowledge. 
+    * They should be using some common code editor/IDE (VSCode/Goland), for the duration of mentorship, so that it is easier to pair program and share knowledge with them, without being blocked by things like code navigation.
+  * It is preferable to have mentees from historically underrepresented communities in tech, to ensure diverse contributors in Thanos.
```

Comment:
:+1:

Reply:
Can we put this as the first item? Quite important.

Reply:
Moved to top

Reply:
Personally I would not put too much emphasis on this and make it a nice to have. Otherwise we are discriminating against people who are a good fit for various other reasons. I would not give this more weight than any other factor, and treat it as more of a tie-breaker.